fuel-tests/tests/messages.rs

```diff
@@ -207,3 +232,108 @@ async fn messages_empty_results_for_owner_with_no_messages(
 
     assert_eq!(result.results.len(), 0);
 }
+
+#[tokio::test]
+async fn can_get_message_proof() {
+    let transaction_id: TransactionId = Bytes32::default().into();
+    let message_id: fuel_core::schema::scalars::MessageId =
+        fuel_gql_client::fuel_types::MessageId::default().into();
+    let block_id = Bytes32::default();
+
+    let config = Config::local_node();
+    let mut db = Database::default();
+
+    db.storage::<Receipts>()
+        .insert(
+            &(transaction_id.into()),
+            &[Receipt::message_out(
+                message_id.into(),
+                Default::default(),
+                Default::default(),
+                Default::default(),
+                Default::default(),
+                Default::default(),
+                Default::default(),
+            )],
+        )
+        .unwrap();
+    let outputs = vec![Output::Message {
+        recipient: Default::default(),
+        amount: Default::default(),
+    }];
+    let txn = Transaction::script(
+        Default::default(),
+        Default::default(),
+        Default::default(),
+        Default::default(),
+        Default::default(),
+        Default::default(),
+        outputs,
+        Default::default(),
+    );
+    db.storage::<Transactions>()
+        .insert(&(transaction_id.into()), &txn)
+        .unwrap();
+    db.update_tx_status(
+        &(transaction_id.into()),
+        fuel_core::tx_pool::TransactionStatus::Success {
+            block_id,
+            time: Default::default(),
+            result: fuel_gql_client::state::ProgramState::Return(Default::default()),
+        },
+    )
+    .unwrap();
+    db.storage::<FuelBlocks>()
+        .insert(
+            &block_id,
+            &FuelBlockDb {
+                header: Default::default(),
+                transactions: vec![transaction_id.into()],
+            },
+        )
+        .unwrap();
+    db.storage::<Messages>()
+        .insert(&(message_id.into()), &Message::default())
+        .unwrap();
+
+    // setup server & client
+    let srv = FuelService::from_database(db, config).await.unwrap();
+    let client = FuelClient::from(srv.bound_address);
+
+    let result = client
+        .message_proof(
+            transaction_id.to_string().as_str(),
+            message_id.to_string().as_str(),
+        )
+        .await
+        .unwrap()
+        .unwrap();
+
+    let mut tree = fuel_gql_client::fuel_merkle::binary::in_memory::MerkleTree::new();
+    tree.push(fuel_gql_client::fuel_types::MessageId::from(message_id).as_ref());
+    let (expected_root, expected_set) = tree.prove(0).unwrap();
+    assert_eq!(
+        *fuel_gql_client::fuel_types::Bytes32::from(result.proof_root),
+        expected_root
```

Comment:
should we test that the proof root also matches the block message root?

Reply:
But that would only test we actually passed in the same block on line `286` which has nothing to do with what we are testing.